superbench/benchmarks/micro_benchmarks/computation_communication_overlap.py

```diff
@@ -0,0 +1,242 @@
+            for stage in range(stages):
+                if kernel == ComputationKernelType.MUL:
+                    matA[stage].mul(matA[stage])
+                elif kernel == ComputationKernelType.MATMUL:
+                    matA[stage].matmul(matB)
+                else:
+                    logger.error(
+                        'Unknown comoputation kernel type - benchmark: {}, type: {}.'.format(self._name, kernel)
+                    )
+                    return False
+        return True
+
+    def _benchmark(self):
+        """Implementation for benchmarking."""
+        M = self._args.m
+        K = self._args.k
+        N = self._args.n
+        P = self._args.p
+        Q = self._args.q
+        kernel = self._args.kernel
+        if self.__local_rank == 0:
+            logger.info(
+                'Computation Communication Overlap - using {} GPUs,\
+                matrix shape for computation: M={} K={} N={},\
+                message tensor shape of nccl = [{},{}],\
+                ratio between computation kernel and nccl kernel={}'.format(
+                    self.__world_size, M, K, N, P, Q, self._args.ratio
+                )
+            )
+
+        MatA = list()
+        MatB = list()
+        # Matrix A
+        for _ in range(self._args.ratio):
+            MatA.append(torch.randn(M, K).cuda())
+        # Matrix B
+        MatB = torch.randn(K, N).cuda()
+        # message for nccl to transport
+        shape = [P, Q]
+        message = torch.randn(*shape).cuda()
+
+        # warm up
+        for i in range(self._args.num_warmup):
+            if not self.__kernel_nccl_pipeline(kernel, MatA, MatB, self._args.ratio, message, times=100):
+                return False
+        torch.cuda.synchronize()
+
+        # run and collect results
+        start = time.perf_counter()
+        for i in range(self._args.num_steps):
+            self.__kernel_nccl_pipeline(kernel, MatA, MatB, self._args.ratio, message, times=100)
+        compute_end = time.perf_counter()
+        torch.cuda.synchronize()
+        end = time.perf_counter()
+
+        logger.info(
+            'GPU {} total compute cost {}'.format(
+                self.__local_rank, (compute_end - start) * 1000 / self._args.num_steps
+            )
+        )
+
+        compute_metric = 'computation_communication_overlap_gpu{}_computation_cost_{}'.format(self.__local_rank, kernel)
+        total_metric = 'computation_communication_overlap_gpu{}_total_cost_{}'.format(self.__local_rank, kernel)
+        compute_elapse_times = [(compute_end - start) * 1000 / self._args.num_steps]
+        elapse_times = [(end - start) * 1000 / self._args.num_steps]
+        if not self._process_numeric_result(compute_metric, compute_elapse_times
+                                            ) or not self._process_numeric_result(total_metric, elapse_times):
+            return False
+
+        logger.info(
```

Comment:
It's better to log benchmark name (self._name) and the time of run (self._curr_run_index) here. We can config how many times to run the benchmark, self._curr_run_index is the current index of running.